athena-federation-sdk/src/main/java/com/amazonaws/athena/connector/lambda/data/BlockUtils.java

```diff
@@ -455,7 +481,8 @@ public static int copyRows(Block srcBlock, Block dstBlock, int firstRow, int las
             for (int i = firstRow; i <= lastRow; i++) {
                 FieldVector dst = dstBlock.getFieldVector(src.getField().getName());
                 src.setPosition(i);
-                setValue(dst, dstOffset++, src.readObject());
+                Object normalizeObject = normalizeObject(src.readObject());
```

Comment:
this adds an extra comparison on every every cell, can we do this in a way that we don't need to check every cell's type?